pkg/gits/bitbucket_server.go

```diff
@@ -370,7 +373,7 @@ func (b *BitbucketServerProvider) CreatePullRequest(data *GitPullRequestArgument
 			"repository": map[string]interface{}{
 				"slug": data.GitRepository.Name,
 				"project": map[string]interface{}{
-					"key": data.GitRepository.Project,
+					"key": strings.ToUpper(data.GitRepository.Organisation),
```

Comment:
Why the need for capitalization? Maybe a comment would be nice somewhere why this is needed.
That said, I find this interesting. Here seems to be a provider (kind) specific behaviour again. If the org in case of BitBucket always needs to be upper-cased, it seems to be quite fragile having to remember to upper case everywhere where `data.GitRepository.Organisation` is used. Really `GitRepository` should either be aware of, or as with the HTTPCloneURL the `GitKind` should know how to handle this.

Reply:
So I’m not 100% sure the API calls will fail when using lowercase for owner, but I am 100% sure that the BitBucket Server “project” will always be uppercase, just like JIRA projects or Confluence spaces. 

My changes regarding case in this PR are generally defensive (I know upper case will always work, so let’s do that) and in this case, `Project` is only relevant for BBS, so it definitely should be uppercase. I opted not to uppercase the values for the `Organisation` field because that one’s used across all providers...though it was made more confusing in that the BBS provider sometimes used `Project` where every other provider would use `Organisation`. Fun times.

Reply:
> So I’m not 100% sure the API calls will fail when using lowercase for owner, but I am 100% sure that the BitBucket Server “project” will always be uppercase, just like JIRA projects or Confluence spaces.

So, do we have a (BDD) testbed for this? Are you just developing against a temporary installation you have somewhere? Could we get BitBucket covered via BDD? That would by far be the best.

Reply:
Yup, https://github.com/jenkins-x/lighthouse/pull/737 has me exercising the Lighthouse suite against BBS.